presto-main/src/main/java/com/facebook/presto/cost/FilterStatsCalculator.java

```diff
@@ -423,14 +415,13 @@ private Type getType(Expression expression)
                                 // At this stage, there should be no subqueries in the plan.
                                 node -> new IllegalStateException("Unexpected Subquery"),
                                 false);
-                        Type type = expressionAnalyzer.analyze(expression, Scope.create());
-                        return type;
```

Comment:
just FYI: sometimes when i debug the code, i found seemingly redundant assignment before return very convenient. I know Intellij can help me if there was no assinment, but it's not nearly as convenient.

(i don't mind your change here though, this is only FYI)